rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,202 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.chain.ethereum.modules.uniswap.v2.constants import (
+    SWAP_SIGNATURE as UNISWAP_V2_SWAP_SIGNATURE,
+)
+from rotkehlchen.chain.ethereum.modules.uniswap.v3.constants import DIRECT_SWAP_SIGNATURE
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import BUY_SIGNATURE, CPT_PARASWAP, SWAP_SIGNATURE as PARASWAP_SWAP_SIGNATURE
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ParaswapCommonDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            router_address: ChecksumEvmAddress,
+            fee_receiver_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(evm_inquirer, base_tools, msg_aggregator)
+        self.router_address = router_address
+        self.fee_receiver_address = fee_receiver_address
+
+    def _decode_swap(
+            self,
+            context: DecoderContext,
+            receiver: ChecksumEvmAddress,
+            sender: ChecksumEvmAddress,
+    ) -> DecodingOutput:
+        """This function is used to decode the swap done by paraswap.
+        At the moment it does not decode the fee in ETH. For example:
+        etherscan.io/tx/0xeeea20b39f157fe59fa4904fd4b62f8971188b53d05c6831e0ed67ee157e40c2"""
+        if not self.base.is_tracked(sender) and not self.base.is_tracked(receiver):
```

Comment:
you case use `any_tracked()`
```suggestion
        if not self.base.any_tracked([sender, receiver]):
```